### PR TITLE
Issue #991: Add data versioning to content blocks API

### PR DESCRIPTION
**Pull request by Vinai**
This is step 1 the remaining 3 of issue #991.

### CODE REVIEW

File: tests/Unit/Suites/Util/FunctionsTest.php [hunk 1]
```diff
@@ -108,6 +115,40 @@ public function typeofDataProvider() : array
...
+    public function testPipelineSingleFunction()
+    {
+        $this->assertSame('FOO BAR', pipeline('strtoupper')('foo bar'));
+    }
+    
+    public function testPipelineCallsFunctionsLeftToRight()
+    {
+        $r = pipeline('strtolower', 'ucwords', function ($str) { return explode(' ', $str); }, 'array_reverse');
+        $this->assertTrue(is_callable($r));
+        $this->assertSame(['Bar', 'Foo'], $r('fOO BaR'));
```
Comment (tim-bezhashvyly) [inline tests/Unit/Suites/Util/FunctionsTest.php]: That's awesome! 😲

File: src/Import/ContentBlock/RestApi/ContentBlocksApiV2PutRequestHandler.php [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+    private function buildKeyGeneratorParamsFromRequest($requestBody): array
+    {
+        $keyGeneratorParams = [];
+
+        if (isset($requestBody['url_key'])) {
+            $keyGeneratorParams['url_key'] = $requestBody['url_key'];
+        }
+
+        return $keyGeneratorParams;
```
Comment (Vinai) [inline src/Import/ContentBlock/RestApi/ContentBlocksApiV2PutRequestHandler.php]: No, I like it better, thanks! In fact, I like a ternary even more :D

File: src/Util/functions.php [hunk 3]
```diff
@@ -28,3 +28,15 @@ function shutdown(int $exitCode = null)
...
+/**
+ * @experimental Please tell us if you use this function. Otherwise we might remove it again, should we not use it.
+ */
+function pipeline(callable $f, callable ...$fs): callable
+{
+    return array_reduce($fs, function (callable $acc, callable $f): callable {
+        return function (...$args) use ($acc, $f) {
+            return $f($acc(...$args));
+        };
+    }, $f);
```
Comment (Vinai) [inline src/Util/functions.php]: Hmm, okay. Changed to `$initialFunction` and `$otherFunctions`.

File: tests/Unit/Suites/Util/FunctionsTest.php (hunk 1 above)
Comment (Vinai) [inline tests/Unit/Suites/Util/FunctionsTest.php]: The hopefully upcoming short array syntax for closures will make that very nice.
Also relevant here: https://wiki.php.net/rfc/pipe-operator

**Comment by Vinai** [comment]
Ready for next review.

**Pull request merged**